cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -506,98 +429,129 @@ func (s *Server) setupRepoKubeappsManagedSecrets(
+	auth *corev1.PackageRepositoryAuth,
+	existingSecretRef *fluxmeta.LocalObjectReference) (string, error) {
+
+	secret, err := newSecretFromTlsConfigAndAuth(repoName, tlsConfig, auth)
+	if err != nil {
+		return "", err
 	}
 
-	if _, ok := secret.Data["caFile"]; ok {
-		tlsConfig = &corev1.PackageRepositoryTlsConfig{
-			// flux plug in doesn't support this option
-			InsecureSkipVerify: false,
-			PackageRepoTlsConfigOneOf: &corev1.PackageRepositoryTlsConfig_SecretRef{
-				SecretRef: &corev1.SecretKeyReference{
-					Name: secret.Name,
-					Key:  "caFile",
-				},
-			},
-		}
+	secretRef := ""
+	typedClient, err := s.clientGetter.Typed(ctx, s.kubeappsCluster)
+	if err != nil {
+		return "", err
 	}
-	if _, ok := secret.Data["certFile"]; ok {
-		if _, ok = secret.Data["keyFile"]; ok {
-			auth.Type = corev1.PackageRepositoryAuth_PACKAGE_REPOSITORY_AUTH_TYPE_TLS
-			auth.PackageRepoAuthOneOf = &corev1.PackageRepositoryAuth_SecretRef{
-				SecretRef: &corev1.SecretKeyReference{Name: secret.Name},
+	secretInterface := typedClient.CoreV1().Secrets(repoName.Namespace)
+	if secret != nil {
+		if existingSecretRef == nil {
+			// create a secret first
+			newSecret, err := secretInterface.Create(ctx, secret, metav1.CreateOptions{})
+			if err != nil {
+				return "", statuserror.FromK8sError("create", "secret", secret.GetGenerateName(), err)
 			}
-		}
-	} else if _, ok := secret.Data["username"]; ok {
-		if _, ok = secret.Data["password"]; ok {
-			auth.Type = corev1.PackageRepositoryAuth_PACKAGE_REPOSITORY_AUTH_TYPE_BASIC_AUTH
-			auth.PackageRepoAuthOneOf = &corev1.PackageRepositoryAuth_SecretRef{
-				SecretRef: &corev1.SecretKeyReference{Name: secret.Name},
+			secretRef = newSecret.GetName()
+		} else {
+			// TODO (gfichtenholt) we should optimize this to somehow tell if the existing secret
+			// is the same (data-wise) as the new one and if so skip all this
+			if err = secretInterface.Delete(ctx, existingSecretRef.Name, metav1.DeleteOptions{}); err != nil {
+				return "", statuserror.FromK8sError("get", "secret", existingSecretRef.Name, err)
 			}
+			// create a new one
+			newSecret, err := secretInterface.Create(ctx, secret, metav1.CreateOptions{})
+			if err != nil {
+				return "", statuserror.FromK8sError("update", "secret", secret.GetGenerateName(), err)
+			}
+			secretRef = newSecret.GetName()
 		}
-	} else if _, ok := secret.Data[".dockerconfigjson"]; ok {
-		auth.Type = corev1.PackageRepositoryAuth_PACKAGE_REPOSITORY_AUTH_TYPE_DOCKER_CONFIG_JSON
-		auth.PackageRepoAuthOneOf = &corev1.PackageRepositoryAuth_SecretRef{
-			SecretRef: &corev1.SecretKeyReference{Name: secret.Name},
+	} else if existingSecretRef != nil {
+		if err = secretInterface.Delete(ctx, existingSecretRef.Name, metav1.DeleteOptions{}); err != nil {
+			log.Errorf("Error deleting existing secret: [%s] due to %v", err)
 		}
-	} else {
-		log.Warning("Unrecognized type of secret [%s]", secret.Name)
 	}
-	return tlsConfig, auth, nil
+	return secretRef, nil
```

Comment:
Confused at first because I thought a secret reference was an object (which wouldn't match the function return type), but see above it's just the name of the secret. No change needed, just noting.